sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -326,9 +326,14 @@ case class UnresolvedExtractValue(child: Expression, extraction: Expression)
  *
  * @param child The computation that is needs to be resolved during analysis.
  * @param aliasName The name if specified to be associated with the result of computing [[child]]
+ * @param aliasFunc The function if specified to be called to generate an alias to associate
+ *                  with the result of computing [[child]]
  *
  */
-case class UnresolvedAlias(child: Expression, aliasName: Option[String] = None)
+case class UnresolvedAlias(
+    child: Expression,
+    aliasName: Option[String] = None,
+    aliasFunc: Option[Expression => String] = None)
```

Comment:
I think `aliasFunc` is enough. Ideally `aliasName` is also a function that ignore the parameter and return a string.